x/fees/keeper/msg_server.go

```diff
@@ -19,61 +20,79 @@ func (k Keeper) RegisterFee(
 	msg *types.MsgRegisterFee,
 ) (*types.MsgRegisterFeeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	if !k.isEnabled(ctx) {
-		return nil, sdkerrors.Wrapf(types.ErrInternalFee, "fees module is not enabled")
+
+	params := k.GetParams(ctx)
+	if !params.EnableFees {
+		return nil, types.ErrFeesDisabled
 	}
 
 	contract := common.HexToAddress(msg.ContractAddress)
-	if k.IsFeeRegistered(ctx, contract) {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "contract is already registered %s", contract)
+
+	if _, found := k.GetFee(ctx, contract); found {
```

Comment:
if you are not using the value (in this case `Fee`), it's better if you create a `HasFee(ctx, contract)` function because it has a constant gas consumption as opposed to the constant + per byte gas rate.